plugins/modules/s3_lifecycle.py

```diff
@@ -10,105 +10,110 @@
+      - Whether to replace all the current transition(s) with the new transition(s).
+      - When false, the provided transition(s) will be added, replacing transitions
+        with the same storage_class. When true, existing transitions will be removed
+        and replaced with the new transition(s)
     default: true
     type: bool
   noncurrent_version_expiration_days:
     description:
-      - 'Delete noncurrent versions this many days after they become noncurrent'
+      - The number of days after which non-current versions should be deleted.
     required: false
     type: int
   noncurrent_version_storage_class:
     description:
-      - 'Transition noncurrent versions to this storage class'
+      - The storage class to which non-current versions are transitioned.
     default: glacier
     choices: ['glacier', 'onezone_ia', 'standard_ia', 'intelligent_tiering', 'deep_archive']
     required: false
     type: str
   noncurrent_version_transition_days:
     description:
-      - 'Transition noncurrent versions this many days after they become noncurrent'
+      - The number of days after which non-current versions will be transitioned
+        to the storage class specified in I(noncurrent_version_storage_class).
     required: false
     type: int
   noncurrent_version_transitions:
     description:
-      - >
-        A list of transition behaviors to be applied to noncurrent versions for the rule. Each storage class may be used only once. Each transition
-        behavior contains these elements
+      - A list of transition behaviors to be applied to noncurrent versions for the rule.
+      - Each storage class may be used only once. Each transition behavior contains these elements
           I(transition_days)
           I(storage_class)
     type: list
     elements: dict
   rule_id:
     description:
-      - "Unique identifier for the rule. The value cannot be longer than 255 characters. A unique value for the rule will be generated if no value is provided."
+      - Unique identifier for the rule.
+      - The value cannot be longer than 255 characters.
+      - A unique value for the rule will be generated if no value is provided.
     type: str
   state:
     description:
-      - "Create or remove the lifecycle rule"
+      - Create or remove the lifecycle rule.
     default: present
     choices: [ 'present', 'absent' ]
     type: str
   status:
     description:
-      - "If 'enabled', the rule is currently being applied. If 'disabled', the rule is not currently being applied."
+      - If C(enabled), the rule is currently being applied.
+      - If C(disabled), the rule is not currently being applied.
     default: enabled
     choices: [ 'enabled', 'disabled' ]
     type: str
   storage_class:
     description:
-      - "The storage class to transition to."
-      - "The 'standard_ia' class is only being available from Ansible version 2.2."
+      - The storage class to transition to.
```

Comment:
Could you also update the available storage classes? https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.upload_fileobj

Reply:
Which class is missing?  I only see the same 5 as listed in choices

Reply:
My bad, sorry! You're right.

Reply:
Should anyone else be looking: https://github.com/boto/botocore/blob/develop/botocore/data/s3/2006-03-01/service-2.json#L9233  (TransitionStorageClass) will tell you what is supported by botocore